divi/src/main.cpp

```diff
@@ -193,24 +193,24 @@ std::set<int> setDirtyFileInfo;
 NotificationInterfaceRegistry registry;
 MainNotificationSignals& g_signals = registry.getSignals();
 
-void RegisterValidationInterface(NotificationInterface* pwalletIn)
+void RegisterValidationInterface(NotificationInterfaceRegistry * registry,NotificationInterface* pwalletIn)
 {
-    registry.RegisterValidationInterface(pwalletIn);
+    registry->RegisterValidationInterface(pwalletIn);
 }
 
-void UnregisterValidationInterface(NotificationInterface* pwalletIn)
+void UnregisterValidationInterface(NotificationInterfaceRegistry * registry,NotificationInterface* pwalletIn)
 {
-    registry.UnregisterValidationInterface(pwalletIn);
+    registry->UnregisterValidationInterface(pwalletIn);
 }
 
-void UnregisterAllValidationInterfaces()
+void UnregisterAllValidationInterfaces(NotificationInterfaceRegistry * registry)
 {
-    registry.UnregisterAllValidationInterfaces();
+    registry->UnregisterAllValidationInterfaces();
 }
 
-void SyncWithWallets(const CTransaction& tx, const CBlock* pblock)
+void SyncWithWallets(NotificationInterfaceRegistry * registry,const CTransaction& tx, const CBlock* pblock)
 {
```

Comment:
Should pass these by reference instead of by pointer.
Additionally, considering that these functions are all accessing a common global, seems to make more sense to encapsulate these in a class instead of passing the argument to the functions directly.

Reply:
fixed it